TestConvertDuration/invalid_duration failure

## 🐛 Bug Report

Running tests fail here while building 1.14.7 or HEAD on x86_64

## To Reproduce

```
$ git clone git@github.com:grpc-ecosystem/grpc-gateway.git
Cloning into 'grpc-gateway'...
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 9811 (delta 1), reused 1 (delta 1), pack-reused 9806
Receiving objects: 100% (9811/9811), 3.93 MiB | 9.10 MiB/s, done.
Resolving deltas: 100% (6847/6847), done.

$ cd grpc-gateway/runtime/

$ go test
go: downloading google.golang.org/grpc v1.31.1
go: downloading github.com/golang/protobuf v1.3.3
go: downloading google.golang.org/genproto v0.0.0-20200513103714-09dca8ec2884
--- FAIL: TestConvertDuration (0.00s)
    --- FAIL: TestConvertDuration/invalid_duration (0.00s)
        convert_test.go:129: got unexpected error
            &errors.errorString{s:"bad Duration: time: unknown unit \"years\" in duration \"123years\""}
            expected
            &errors.errorString{s:"bad Duration: time: unknown unit years in duration 123years"}
FAIL
exit status 1
FAIL	github.com/grpc-ecosystem/grpc-gateway/runtime	0.015s
```

## Your Environment

Mageia Linux

go version go1.15 linux/amd64

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -41,7 +41,7 @@ executors:
       password1: '3ec86b2e5a431be2d72c'
       GLOG_logtostderr: '1'
     docker:
-      - image: docker.pkg.github.com/grpc-ecosystem/grpc-gateway/build-env:1.14
+      - image: docker.pkg.github.com/grpc-ecosystem/grpc-gateway/build-env:1.15
         auth:
           username: gateway-ci-user
           password: ${password0}${password1}

--- a/runtime/convert_test.go
+++ b/runtime/convert_test.go
@@ -1,9 +1,6 @@
 package runtime_test
 
 import (
-	"encoding/json"
-	"fmt"
-	"reflect"
 	"testing"
 
 	"github.com/golang/protobuf/proto"
@@ -17,7 +14,7 @@ func TestConvertTimestamp(t *testing.T) {
 		name    string
 		input   string
 		output  *timestamp.Timestamp
-		wanterr error
+		wanterr bool
 	}{
 		{
 			name:  "a valid RFC3339 timestamp",
@@ -26,46 +23,37 @@ func TestConvertTimestamp(t *testing.T) {
 				Seconds: 1462875553,
 				Nanos:   123000000,
 			},
-			wanterr: nil,
+			wanterr: false,
 		},
 		{
 			name:    "invalid timestamp",
 			input:   `"05-10-2016T10:19:13.123Z"`,
 			output:  nil,
-			wanterr: fmt.Errorf(`bad Timestamp: parsing time "05-10-2016T10:19:13.123Z" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "0-2016T10:19:13.123Z" as "2006"`),
+			wanterr: true,
 		},
 		{
-			name:   "JSON number",
-			input:  "123",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "number",
-				Type:   reflect.TypeOf("123"),
-				Offset: 3,
-			},
+			name:    "JSON number",
+			input:   "123",
+			output:  nil,
+			wanterr: true,
 		},
 		{
-			name:   "JSON bool",
-			input:  "true",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "bool",
-				Type:   reflect.TypeOf("123"),
-				Offset: 4,
-			},
+			name:    "JSON bool",
+			input:   "true",
+			output:  nil,
+			wanterr: true,
 		},
 	}
 
 	for _, spec := range specs {
 		t.Run(spec.name, func(t *testing.T) {
 			ts, err := runtime.Timestamp(spec.input)
-			if spec.wanterr != nil {
-				if !reflect.DeepEqual(err, spec.wanterr) {
-					t.Errorf("got unexpected error\n%#v\nexpected\n%#v", err, spec.wanterr)
-				}
-				return
-			}
-			if !proto.Equal(ts, spec.output) {
+			switch {
+			case err != nil && !spec.wanterr:
+				t.Errorf("got unexpected error\n%#v", err)
+			case err == nil && spec.wanterr:
+				t.Errorf("did not error when expecte")
+			case !proto.Equal(ts, spec.output):
 				t.Errorf(
 					"when testing %s; got\n%#v\nexpected\n%#v",
 					spec.name,
@@ -82,7 +70,7 @@ func TestConvertDuration(t *testing.T) {
 		name    string
 		input   string
 		output  *duration.Duration
-		wanterr error
+		wanterr bool
 	}{
 		{
 			name:  "a valid duration",
@@ -91,46 +79,37 @@ func TestConvertDuration(t *testing.T) {
 				Seconds: 123,
 				Nanos:   456000000,
 			},
-			wanterr: nil,
+			wanterr: false,
 		},
 		{
 			name:    "invalid duration",
 			input:   `"123years"`,
 			output:  nil,
-			wanterr: fmt.Errorf(`bad Duration: time: unknown unit years in duration 123years`),
+			wanterr: true,
 		},
 		{
-			name:   "JSON number",
-			input:  "123",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "number",
-				Type:   reflect.TypeOf("123"),
-				Offset: 3,
-			},
+			name:    "JSON number",
+			input:   "123",
+			output:  nil,
+			wanterr: true,
 		},
 		{
-			name:   "JSON bool",
-			input:  "true",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "bool",
-				Type:   reflect.TypeOf("123"),
-				Offset: 4,
-			},
+			name:    "JSON bool",
+			input:   "true",
+			output:  nil,
+			wanterr: true,
 		},
 	}
 
 	for _, spec := range specs {
 		t.Run(spec.name, func(t *testing.T) {
 			ts, err := runtime.Duration(spec.input)
-			if spec.wanterr != nil {
-				if !reflect.DeepEqual(err, spec.wanterr) {
-					t.Errorf("got unexpected error\n%#v\nexpected\n%#v", err, spec.wanterr)
-				}
-				return
-			}
-			if !proto.Equal(ts, spec.output) {
+			switch {
+			case err != nil && !spec.wanterr:
+				t.Errorf("got unexpected error\n%#v", err)
+			case err == nil && spec.wanterr:
+				t.Errorf("did not error when expecte")
+			case !proto.Equal(ts, spec.output):
 				t.Errorf(
 					"when testing %s; got\n%#v\nexpected\n%#v",
 					spec.name,

